packages/core/saved-objects/core-saved-objects-api-server-internal/src/lib/repository.ts

```diff
@@ -1292,23 +1605,44 @@ export class SavedObjectsRepository implements ISavedObjectsRepository {
     objects: SavedObjectsBulkGetObject[] = [],
     options: SavedObjectsBaseOptions = {}
   ): Promise<SavedObjectsBulkResponse<T>> {
-    const namespace = normalizeNamespace(options.namespace);
+    const namespace = this.getCurrentNamespace(options.namespace);
 
     if (objects.length === 0) {
       return { saved_objects: [] };
     }
 
+    let availableSpacesPromise: Promise<string[]> | undefined;
+    const getAvailableSpaces = async () => {
+      if (!availableSpacesPromise) {
+        // This function is only called below if the spaces extension is enabled, so we can safely use the non-null assertion operator here.
```

Comment:
I think it's preferable to pass in the spacesExtension as a required param (that cannot be undefined). That way we can avoid the non-null assertion. I think it's also worth breaking this function out of the scope of the parent function and instead passing in all the state with paramaters.

Reply:
@azasypkin Any opposition to requiring the spaces extension?

Reply:
> @azasypkin Any opposition to requiring the spaces extension?

Assuming you mean passing `spacesExtensions` to the _internal_ `getAvailableSpaces` function as a required argument - no objections from my side.

Reply:
Thanks. I have refactored this.

@rudolf Just for my own understanding, how expensive is the non-null assertion? Is there a strong reason to avoid this?

Reply:
@rudolf I am actually inclined to move the function back into the bulkGet function - it is the only use, and the return value for 403/not authorized is specific to the flow of bulkGet. The two other use cases of spacesExtension.getSearchableNamespaces also have a specific return/throw for a 403 status.

Reply:
> how expensive is the non-null assertion?

This is to improve type safety not for performance.